src/bindings/c/include/openvino/c/ov_core.h

```diff
@@ -0,0 +1,268 @@
+    const char* description;  //!< A string representing OpenVINO description
+} ov_version_t;
+
+/**
+ * @struct ov_core_version
+ * @brief  Represents version information that describes device and ov runtime library
+ */
+typedef struct {
+    const char* device_name;  //!< A device name
+    ov_version_t version;     //!< Version
+} ov_core_version_t;
+
+/**
+ * @struct ov_core_version_list
+ * @brief  Represents version information that describes all devices and ov runtime library
+ */
+typedef struct {
+    ov_core_version_t* versions;  //!< An array of device versions
+    size_t size;                  //!< A number of versions in the array
+} ov_core_version_list_t;
+
+/**
+ * @struct ov_available_devices_t
+ * @brief Represent all available devices.
+ */
+typedef struct {
+    char** devices;
+    size_t size;
+} ov_available_devices_t;
+
+/**
+ * @brief Get version of OpenVINO.
+ * @param ov_version_t a pointer to the version
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_get_openvino_version(ov_version_t* version);
+
+/**
+ * @brief Release the memory allocated by ov_version_t.
+ * @param version A pointer to the ov_version_t to free memory.
+ */
+OPENVINO_C_API(void) ov_version_free(ov_version_t* version);
+
+// OV Core
+
+/**
+ * @defgroup Core Core
+ * @ingroup openvino_c
+ * Set of functions dedicated to working with registered plugins and loading
+ * model to the registered devices.
+ * @{
+ */
+
+/**
+ * @brief Constructs OpenVINO Core instance by default.
+ * See RegisterPlugins for more details.
+ * @ingroup Core
+ * @param core A pointer to the newly created ov_core_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_create(ov_core_t** core);
+
+/**
+ * @brief Constructs OpenVINO Core instance using XML configuration file with devices description.
+ * See RegisterPlugins for more details.
+ * @ingroup Core
+ * @param xml_config_file A path to .xml file with devices to load from. If XML configuration file is not specified,
+ * then default plugin.xml file will be used.
+ * @param core A pointer to the newly created ov_core_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_create_with_config(const char* xml_config_file, ov_core_t** core);
+
+/**
+ * @brief Release the memory allocated by ov_core_t.
+ * @ingroup Core
+ * @param core A pointer to the ov_core_t to free memory.
+ */
+OPENVINO_C_API(void) ov_core_free(ov_core_t* core);
+
+/**
+ * @brief Reads models from IR/ONNX/PDPD formats.
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_path Path to a model.
+ * @param bin_path Path to a data file.
+ * For IR format (*.bin):
+ *  * if path is empty, will try to read a bin file with the same name as xml and
+ *  * if the bin file with the same name is not found, will load IR without weights.
```

Comment:
am I right that I can pass bin_path as NULL and OpenVINO will detect the bin file automatically? Let's add this to the docs then

Reply:
It should be C++ api logic.